core/ir/x86/decode.c

```diff
@@ -2762,6 +2755,15 @@ byte *
 decode(void *drcontext, byte *pc, instr_t *instr)
 {
     dcontext_t *dcontext = (dcontext_t *)drcontext;
+
+    /* Synthetic ISA has its own decoder.
+     * XXX i#1684: when DR can be built with full dynamic architecture selection we won't
+     * need to pollute the decoding of other architectures with this synthetic ISA special
+     * case.
+     */
+    if (instr_get_isa_mode(instr) == DR_ISA_SYNTHETIC)
```

Comment:
See other comment: we want to use the dcontext mode here.

Reply:
Yes, done. https://github.com/DynamoRIO/dynamorio/pull/6691#discussion_r1533506865